### PR TITLE
Introduce recursive project parsing.

### PR DESCRIPTION
**Pull request by 4383**
- introduce the Module class
- refactor the Analyze class
- introduce global stats
- accept file and dir
- check if the file is a valid python file
- adding unittests

### CODE REVIEW

**Comment by 4383** [comment]
#5

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,4 @@ dist/
 *.swp
 *~
 Pipfile
+.coverage
```
Comment (hobbestigrou) [inline .gitignore]: I think it's better to make this in another patch.

File: kojak/__main__.py [hunk 2]
```diff
@@ -1,66 +1,39 @@
-import argparse
 from sys import exit
 
+import kojak.cli as cli
```
Comment (hobbestigrou) [inline kojak/__main__.py]: Why do you use as here? A simple import is sufficient.

File: kojak/cli.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
+import argparse
+import os
+
+
+def is_valid_path(path):
+    if not os.path.isdir(path) and not os.path.isfile(path):
```
Comment (hobbestigrou) [inline kojak/cli.py]: I don't understand this condition? How that can be not and file and not a directory?

File: test-requirements.txt [hunk 4]
```diff
@@ -3,3 +3,5 @@ pbr
 tox
 bandit
 nose
+coverage
+nose-timer
```
Comment (hobbestigrou) [inline test-requirements.txt]: I think it's better to make this in another patch.

File: tox.ini [hunk 5]
```diff
@@ -5,7 +5,10 @@ envlist = py34,py35,py36,pep8
 usedevelop = True
 install_command = pip install {opts} {packages}
 deps = -r{toxinidir}/test-requirements.txt
-commands = nosetests
+commands =
+    nosetests \
+    --with-timer \
+    --with-coverage --cover-erase --cover-package=kojak
```
Comment (hobbestigrou) [inline tox.ini]: I think it's better to make this in another patch.

File: kojak/cli.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    stats = analyze.get_global_stats()
+    print('This project contains {count} {word_class}'.format(
+        count=stats['modules'],
+        word_class=pluralize(stats['modules'], 'module')))
+    print('This project contains {count} {word_class}'.format(
+        count=stats['imports'],
+        word_class=pluralize(stats['imports'], 'import')))
+    print('This project contains {count} {word_class}'.format(
+        count=stats['modules'],
+        word_class=pluralize(stats['classes'], 'class', 'classes')))
```
Comment (hobbestigrou) [inline kojak/cli.py:105]: Why don't use a class for this module?

**Review by hobbestigrou — CHANGES_REQUESTED** [review]

File: kojak/__main__.py (hunk 2 above)
Comment (4383) [reply kojak/__main__.py]: I want to avoid calls by using long namespaces etc...

File: kojak/cli.py (hunk 6 above)
Comment (4383) [reply kojak/cli.py:105]: I don't understand your comment, I just rewrite your code to call the pluralize function and avoid to check this kind of things at many places in kojak

File: kojak/cli.py (hunk 3 above)
Comment (4383) [reply kojak/cli.py]: if the given path is not a dir or not a file in the FS then something went wrong and the path doesn't exist. 

It's ok?

File: .gitignore (hunk 1 above)
Comment (4383) [reply .gitignore]: see #20

File: test-requirements.txt (hunk 4 above)
Comment (4383) [reply test-requirements.txt]: see #20

File: tox.ini (hunk 5 above)
Comment (4383) [reply tox.ini]: see #20

File: kojak/__main__.py (hunk 2 above)
Comment (hobbestigrou) [reply kojak/__main__.py]: Sorry I don't understand what you mean with an import from you the same result.

**Review by hobbestigrou — CHANGES_REQUESTED** [review]

File: kojak/cli.py (hunk 3 above)
Comment (hobbestigrou) [reply kojak/cli.py]: So you can use exists function.

**Review by hobbestigrou — CHANGES_REQUESTED** [review]

**Comment by hobbestigrou** [comment]
Ok for me, I agree.

**Review by hobbestigrou — APPROVED** [review]
